infra/aws/terraform/prow-build-cluster/prow.tf

```diff
@@ -55,5 +55,32 @@ resource "aws_iam_role" "eks_admin" {
   })
 
   max_session_duration = 43200
+}
 
+# We allow ESO pods in the Prow control plane cluster to read from AWS Secrets Manager.
+resource "aws_iam_role" "eso_read" {
+  name = "Prow-ESO-Read"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Federated" : aws_iam_openid_connect_provider.k8s_prow.arn
+        },
+        "Action" : "sts:AssumeRoleWithWebIdentity",
+        "Condition" : {
+          "StringEquals" : {
+            "container.googleapis.com/v1/projects/k8s-prow/locations/us-central1-f/clusters/prow:sub" : [
+              // https://github.com/kubernetes/test-infra/tree/master/config/prow/cluster 
+              // all services that load kubeconfig should be listed here
+              "system:serviceaccount:default:kubernetes-external-secrets-sa"
+            ]
```

Comment:
Can you add a policy for it read SSM?

https://github.com/external-secrets/kubernetes-external-secrets#add-a-secret

```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": [
        "secretsmanager:GetResourcePolicy",
        "secretsmanager:GetSecretValue",
        "secretsmanager:DescribeSecret",
        "secretsmanager:ListSecretVersionIds"
      ],
      "Resource": [
        "arn:aws:secretsmanager:us-west-2:111122223333:secret:aes128-1a2b3c",
        "arn:aws:secretsmanager:us-west-2:111122223333:secret:aes192-4D5e6F",
        "arn:aws:secretsmanager:us-west-2:111122223333:secret:aes256-7g8H9i"
      ]
    }
  ]
}
```